index.js

```diff
@@ -170,35 +170,32 @@ function fastifyView (fastify, opts, next) {
     return viewExt ? `.${viewExt}` : (includeViewExtension ? `.${extension}` : filextension)
   }
 
-  function isPathExcludedMinification (currentPath, pathsToExclude) {
-    return (pathsToExclude && Array.isArray(pathsToExclude)) ? pathsToExclude.includes(currentPath) : false
-  }
+  const minify = typeof globalOptions.useHtmlMinifier?.minify === 'function' &&
+    globalOptions.useHtmlMinifier.minify
 
-  function useHtmlMinification (globalOpts, requestedPath) {
-    return globalOptions.useHtmlMinifier &&
-      (typeof globalOptions.useHtmlMinifier.minify === 'function') &&
-      !isPathExcludedMinification(requestedPath, globalOptions.pathsToExcludeHtmlMinifier)
-  }
+  const minifyExcludedPaths = Array.isArray(globalOptions.pathsToExcludeHtmlMinifier)
+    ? new Set(globalOptions.pathsToExcludeHtmlMinifier)
+    : null
 
   function getRequestedPath (fastify) {
-    return (fastify && fastify.request) ? fastify.request.routeOptions.url : null
+    return fastify?.request?.routeOptions?.url ?? null
```

Comment:
nit, but, it is recommended by the v8 team not to abuse the `?`

[> Still, be considerate about using more than one optional chaining operator in a single chain. If a value is guaranteed to not be nullish, then using ?. to access properties on it is discouraged.](https://v8.dev/features/optional-chaining)

```suggestion
    return fastify?.request?.routeOptions.url ?? null
```